[2025-07-25 Fiddler](https://thefiddler.substack.com/p/can-you-sprint-to-the-finish)
====================

Fiddler
-------
My initial thought is to always sprint.  If we are both under 50%, I win.  If I am under 50% and
my opponent is over 50%, I lose.  If I am over 50% and my opponent is under 50%, I win.  If we
are both over 50%, it's 50-50.  Each of these cases have 25% probability, so my chances of winning
is 5/8.

I don't see how not sprinting in any of the cases improves my chances, not that I'd be able to
distinguish between the cases of my opponent being over or under 50%.

Extra credit
------------
Let $x$ and $y$ be my opponents' legs.  Let $z$ be my legs, and $s$ be my strategy, where
I sprint if $z > s$.  There are $4! = 24$ ways these can be ordered, and each ordering is
equally probable.

Since my opponents are interchangeable, let $x < y$, and there are 12 equally probable
orderings.

There is also the 50% to account for, so bring on the integrals.

In [1]:
x,y,z,s = var("x,y,z,s")

#### When $x < y < z < s$
My chances of winning is $1/3$ when $y < 1/2$ and zero otherwise.

In [2]:
if False:
    # the computer cannot handle this integral
    pxyzs = integral(integral(integral(integral(1/3, x, 0, y),
                                       y, 0, min_symbolic(z,1/2)),
                              z, 0, s),
                     s, 0, 1)*2

pxyzs = (integral(integral(integral(integral(1/3, x, 0, y),
                                    y, 0, z),
                           z, 0, s),
                  s, 0, 1/2)
         + integral(integral(integral(integral(1/3, x, 0, y),
                                      y, 0, z),
                             z, 0, 1/2)
                    + integral(integral(integral(1/3, x, 0, y),
                                        y, 0, 1/2),
                               z, 1/2, s),
                    s, 1/2, 1)
        )*2
pxyzs

11/576

#### When $x < y < s < z$
I always win this case.

In [3]:
pxysz = integral(integral(integral(integral(1, x, 0, y),
                                   y, 0, s),
                          s, 0, z),
                 z, 0, 1)*2
pxysz

1/12

#### When $x < z < y < s$
My chances of winning is $1/3$ when $y < 1/2$, and zero otherwise.

In [4]:
pxzys = integral(integral(integral(integral(1/3, x, 0, z),
                                   z, 0, y),
                          y, 0, min_symbolic(s,1/2)),
                 s, 0, 1)*2
pxzys

5/576

#### When $x < z < s < y$
My chances of winning is $1/3$ when $y < 1/2$, and zero otherwise.

In [5]:
pxzsy = integral(integral(integral(integral(1/3, x, 0, z),
                                   z, 0, s),
                          s, 0, y),
                 y, 0, 1/2)*2
pxzsy

1/576

#### When $x < s < y < z$
I always win this case.

In [6]:
pxsyz = integral(integral(integral(integral(1, x, 0, s),
                                   s, 0, y),
                          y, 0, z),
                 z, 0, 1)*2
pxsyz

1/12

#### When $x < s < z < y$
I sprint and win when $y < 1/2$, and lose otherwise.

In [7]:
pxszy = integral(integral(integral(integral(1, x, 0, s),
                                   s, 0, z),
                          z, 0, y),
                 y, 0, 1/2)*2
pxszy

1/192

#### When $z < x < y < s$
My chances of winning is $1/3$ when $y < 1/2$, and zero otherwise.

In [8]:
pzxys = integral(integral(integral(integral(1/3, z, 0, x),
                                   x, 0, y),
                          y, 0, min_symbolic(s,1/2)),
                 s, 0, 1)*2
pzxys

5/576

#### When $z < x < s < y$
My chances of winning is $1/3$ when $y < 1/2$, and zero otherwise.

In [9]:
pzxsy = integral(integral(integral(integral(1/3, z, 0, x),
                                   x, 0, s),
                          s, 0, y),
                 y, 0, 1/2)*2
pzxsy

1/576

#### When $z < s < x < y$
My chances of winning is $1/3$ when $y < 1/2$, and zero otherwise.

In [10]:
pzsxy = integral(integral(integral(integral(1/3, z, 0, s),
                                   s, 0, x),
                          x, 0, y),
                 y, 0, 1/2)*2
pzsxy

1/576

#### When $s < x < y < z$
I always sprint and win.

In [11]:
psxyz = integral(integral(integral(integral(1, s, 0, x),
                                   x, 0, y),
                          y, 0, z),
                 z, 0, 1)*2
psxyz

1/12

#### When $s < x < z < y$
I always sprint and win when $y < 1/2$, and lose otherwise.

In [12]:
psxzy = integral(integral(integral(integral(1, s, 0, x),
                                   x, 0, z),
                          z, 0, y),
                 y, 0, 1/2)*2
psxzy

1/192

#### When $s < z < x < y$
I always sprint and win when $y < 1/2$, and lose otherwise.

In [13]:
pszxy = integral(integral(integral(integral(1, s, 0, z),
                                   z, 0, x),
                          x, 0, y),
                 y, 0, 1/2)*2
pszxy

1/192

Adding up,

In [14]:
p = (pxyzs + pxysz + pxzys + pxzsy + pxsyz + pxszy
     + pzxys + pzxsy + pzsxy
     + psxyz + psxzy + pszxy)
p,numerical_approx(p)

(59/192, 0.307291666666667)

### Alternative calculation
Since the slower opponent is somewhat irrelevant, we have $y$, $z$, and $s$, and there are
just $3! = 6$ orderings, though not equally probable.  The probability that $y$ is the faster
of the two opponents is $y$, so the normalized probability distribution is $2y$.

Similarly, how $s$ is ordered is also somewhat irrelevant.  The probability that I sprint
is 50% and my probability distribution is $2z$ when I sprint and $2(1-z)$ when I don't sprint.

In the $1/2$ chance that I don't sprint, and in the $1/4$ chance that neither of my opponents
sprint, I have a $1/3$ chance of winning.  So the overall probability that nobody sprints and
I win is $1/24$.

In the $1/2$ chance that I do sprint, and in the $1/4$ chance that neither of my opponents
sprint, I win.  So the overall probability that I win as the only sprinter is $1/8$.

For the case where I and at least one other opponent sprints, my probability of winning is

$$ \int_{1/2}^1 dy\,2y\int_y^1 dz\,2z = \int_{1/2}^1 dy\,2y(1-y^2) = 1 - 1/2 - 1/4 + 1/32 = 9/32 $$

So the overall probability that I win in a sprint is $9/64$.

And the probability that I win is $1/24+1/8+9/64 = (8+24+27)/192 = 59/192$.

### Simulations
[Simulations](20250725.go) agree.

    $ go run 20250725.go
    2458998/8000000 0.307375
    pxyzs=0.019086
    pxysz=0.083245
    pxzys=0.008643
    pxzsy=0.001727
    pxsyz=0.083553
    pxszy=0.005219
    pzxys=0.008668
    pzxsy=0.001748
    pzsxy=0.001727
    psxyz=0.083348
    psxzy=0.005185
    pszxy=0.005225

In [15]:
print(f"pxyzs={pxyzs} {numerical_approx(pxyzs)}")
print(f"pxysz={pxysz} {numerical_approx(pxysz)}")
print(f"pxzys={pxzys} {numerical_approx(pxzys)}")
print(f"pxzsy={pxzsy} {numerical_approx(pxzsy)}")
print(f"pxsyz={pxsyz} {numerical_approx(pxsyz)}")
print(f"pxszy={pxszy} {numerical_approx(pxszy)}")
print(f"pzxys={pzxys} {numerical_approx(pzxys)}")
print(f"pzxsy={pzxsy} {numerical_approx(pzxsy)}")
print(f"pzsxy={pzsxy} {numerical_approx(pzsxy)}")
print(f"psxyz={psxyz} {numerical_approx(psxyz)}")
print(f"psxzy={psxzy} {numerical_approx(psxzy)}")
print(f"pszxy={pszxy} {numerical_approx(pszxy)}")

pxyzs=11/576 0.0190972222222222
pxysz=1/12 0.0833333333333333
pxzys=5/576 0.00868055555555556
pxzsy=1/576 0.00173611111111111
pxsyz=1/12 0.0833333333333333
pxszy=1/192 0.00520833333333333
pzxys=5/576 0.00868055555555556
pzxsy=1/576 0.00173611111111111
pzsxy=1/576 0.00173611111111111
psxyz=1/12 0.0833333333333333
psxzy=1/192 0.00520833333333333
pszxy=1/192 0.00520833333333333


Additional thoughts
===================
The slower opponent becomes relevant if we want to calculate my average finishing
position.  I'll assume that if one rider sprints and the other two do not, the
sprinter finishes first, and the other two have equal probabilities of finishing
second.

We already have the probability of finishing first, 59/192.

Next, the probability of finishing second,

In [16]:
qxyzs = (integral(integral(integral(integral(1/3, x, 0, y),
                                    y, 0, z),
                           z, 0, s),
                  s, 0, 1/2)
         + integral(integral(integral(integral(1/3, x, 0, y),
                                      y, 0, z),
                             z, 0, 1/2)
                    + integral(integral(integral(1/3, x, 0, y),
                                        y, 0, 1/2)
                               + integral(integral(1/2, x, 0, 1/2),
                                          y, 1/2, z),
                               z, 1/2, s),
                    s, 1/2, 1))*2
qxysz = 0
qxzys = (integral(integral(integral(integral(1/3, x, 0, z),
                                    z, 0, y),
                           y, 0, s),
                  s, 0, 1/2)
         + integral(integral(integral(integral(1/3, x, 0, z),
                                      z, 0, y),
                             y, 0, 1/2)
                    + integral(integral(integral(1/2, x, 0, z),
                                        z, 0, 1/2)
                               + integral(integral(1/2, x, 0, 1/2),
                                          z, 1/2, y),
                               y, 1/2, s),
                    s, 1/2, 1))*2
qxzsy = (integral(integral(integral(integral(1/3, x, 0, z),
                                    z, 0, s),
                           s, 0, y),
                  y, 0, 1/2)
         + integral(integral(integral(integral(1/2, x, 0, z),
                                      z, 0, s),
                             s, 0, 1/2)
                    + integral(integral(integral(1/2, x, 0, z),
                                        z, 0, 1/2)
                               + integral(integral(1/2, x, 0, 1/2),
                                          z, 1/2, s),
                               s, 1/2, y),
                    y, 1/2, 1))*2
qxsyz = 0
qxszy = integral(integral(integral(integral(1, x, 0, s),
                                   s, 0, z),
                          z, 0, y),
                 y, 1/2, 1)*2
qzxys = (integral(integral(integral(integral(1/3, z, 0, x),
                                    x, 0, y),
                           y, 0, s),
                  s, 0, 1/2)
         + integral(integral(integral(integral(1/3, z, 0, x),
                                      x, 0, y),
                             y, 0, 1/2)
                    + integral(integral(integral(1/2, z, 0, x),
                                        x, 0, 1/2),
                               y, 1/2, s),
                    s, 1/2, 1))*2
qzxsy = (integral(integral(integral(integral(1/3, z, 0, x),
                                    x, 0, s),
                           s, 0, y),
                  y, 0, 1/2)
         + integral(integral(integral(integral(1/2, z, 0, x),
                                      x, 0, s),
                             s, 0, 1/2)
                    + integral(integral(integral(1/2, z, 0, x),
                                        x, 0, 1/2),
                               s, 1/2, y),
                    y, 1/2, 1))*2
qzsxy = (integral(integral(integral(integral(1/3, z, 0, s),
                                    s, 0, x),
                           x, 0, y),
                  y, 0, 1/2)
         + integral(integral(integral(integral(1/2, z, 0, s),
                                      s, 0, x),
                             x, 0, 1/2),
                    y, 1/2, 1))*2
qsxyz = 0
qsxzy = integral(integral(integral(integral(1, s, 0, x),
                                   x, 0, z),
                          z, 0, y),
                 y, 1/2, 1)*2
qszxy = integral(integral(integral(integral(1, s, 0, z),
                                   z, 0, x),
                          x, 0, 1/2),
                 y, 1/2, 1)*2
q = qxyzs+qxysz+qxzys+qxzsy+qxsyz+qxszy+qzxys+qzxsy+qzsxy+qsxyz+qsxzy+qszxy
q

11/32

And the probability of finishing third,

In [17]:
rxyzs = (integral(integral(integral(integral(1/3, x, 0, y),
                                    y, 0, z),
                           z, 0, s),
                  s, 0, 1/2)
         + integral(integral(integral(integral(1/3, x, 0, y),
                                      y, 0, z),
                             z, 0, 1/2)
                    + integral(integral(integral(1/3, x, 0, y),
                                        y, 0, 1/2)
                               + integral(integral(1/2, x, 0, 1/2)
                                          + integral(1, x, 1/2, y),
                                          y, 1/2, z),
                               z, 1/2, s),
                    s, 1/2, 1))*2
rxysz = 0
rxzys = (integral(integral(integral(integral(1/3, x, 0, z),
                                    z, 0, y),
                           y, 0, s),
                  s, 0, 1/2)
         + integral(integral(integral(integral(1/3, x, 0, z),
                                      z, 0, y),
                             y, 0, 1/2)
                    + integral(integral(integral(1/2, x, 0, z),
                                        z, 0, 1/2)
                               + integral(integral(1/2, x, 0, 1/2)
                                          + integral(1, x, 1/2, z),
                                          z, 1/2, y),
                               y, 1/2, s),
                    s, 1/2, 1))*2
rxzsy = (integral(integral(integral(integral(1/3, x, 0, z),
                                    z, 0, s),
                           s, 0, y),
                  y, 0, 1/2)
         + integral(integral(integral(integral(1/2, x, 0, z),
                                      z, 0, s),
                             s, 0, 1/2)
                    + integral(integral(integral(1/2, x, 0, z),
                                        z, 0, 1/2)
                               + integral(integral(1/2, x, 0, 1/2)
                                          + integral(1, x, 1/2, z),
                                          z, 1/2, s),
                               s, 1/2, y),
                    y, 1/2, 1))*2
rxsyz = 0
rxszy = 0
rzxys = (integral(integral(integral(integral(1/3, z, 0, x),
                                    x, 0, y),
                           y, 0, s),
                  s, 0, 1/2)
         + integral(integral(integral(integral(1/3, z, 0, x),
                                      x, 0, y),
                             y, 0, 1/2)
                    + integral(integral(integral(1/2, z, 0, x),
                                        x, 0, 1/2)
                               + integral(integral(1, z, 0, x),
                                          x, 1/2, y),
                               y, 1/2, s),
                    s, 1/2, 1))*2
rzxsy = (integral(integral(integral(integral(1/3, z, 0, x),
                                    x, 0, s),
                           s, 0, y),
                  y, 0, 1/2)
         + integral(integral(integral(integral(1/2, z, 0, x),
                                      x, 0, s),
                             s, 0, 1/2)
                    + integral(integral(integral(1/2, z, 0, x),
                                        x, 0, 1/2)
                               + integral(integral(1, z, 0, x),
                                          x, 1/2, s),
                               s, 1/2, y),
                    y, 1/2, 1))*2
rzsxy = (integral(integral(integral(integral(1/3, z, 0, s),
                                    s, 0, x),
                           x, 0, y),
                  y, 0, 1/2)
         + integral(integral(integral(integral(1/2, z, 0, s),
                                      s, 0, x),
                             x, 0, 1/2)
                    + integral(integral(integral(1, z, 0, s),
                                        s, 0, x),
                               x, 1/2, y),
                    y, 1/2, 1))*2
rsxyz = 0
rsxzy = 0
rszxy = integral(integral(integral(integral(1, s, 0, z),
                                   z, 0, x),
                          x, 1/2, y),
                 y, 1/2, 1)*2
r = rxyzs+rxysz+rxzys+rxzsy+rxsyz+rxszy+rzxys+rzxsy+rzsxy+rsxyz+rsxzy+rszxy
p,q,r, p+q+r, p+2*q+3*r, numerical_approx(p+2*q+3*r)

(59/192, 11/32, 67/192, 1, 49/24, 2.04166666666667)

And my finishing position on average is $49/24 \approx 2.04167$.